sklego/transformers.py

```diff
@@ -2,28 +2,27 @@
 from sklearn.utils import check_array, check_X_y
 from sklearn.utils.validation import FLOAT_DTYPES, check_random_state, check_is_fitted
 
+from sklego.common import TrainOnlyTransformerMixin
 
-class RandomAdder(TransformerMixin, BaseEstimator):
+
+class RandomAdder(TrainOnlyTransformerMixin, BaseEstimator):
     def __init__(self, noise=1, random_state=None):
         self.noise = noise
         self.random_state = random_state
 
     def fit(self, X, y):
-
+        super().fit(X, y)
```

Comment:
i might be missing something but why do we **need** to call `fit` on super here?

Reply:
in `super().fit()` we compute the hash on `X`